pkg/sdk/database_role.go

```diff
@@ -0,0 +1,132 @@
+type alterDatabaseRoleOptions struct {
+	alter        bool                     `ddl:"static" sql:"ALTER"`
+	databaseRole bool                     `ddl:"static" sql:"DATABASE ROLE"`
+	IfExists     *bool                    `ddl:"keyword" sql:"IF EXISTS"`
+	name         DatabaseObjectIdentifier `ddl:"identifier"`
+
+	// One of
+	Rename *DatabaseRoleRename `ddl:"list,no_parentheses" sql:"RENAME TO"`
+	Set    *DatabaseRoleSet    `ddl:"list,no_parentheses" sql:"SET"`
+	Unset  *DatabaseRoleUnset  `ddl:"list,no_parentheses" sql:"UNSET"`
+}
+
+type DatabaseRoleRename struct {
+	Name DatabaseObjectIdentifier `ddl:"identifier"`
+}
+
+type DatabaseRoleSet struct {
+	Comment string `ddl:"parameter,single_quotes" sql:"COMMENT"`
+}
+
+type DatabaseRoleUnset struct {
+	Comment bool `ddl:"keyword" sql:"COMMENT"`
+}
+
+// dropDatabaseRoleOptions is based on https://docs.snowflake.com/en/sql-reference/sql/drop-database-role.
+type dropDatabaseRoleOptions struct {
+	drop         bool                     `ddl:"static" sql:"DROP"`
+	databaseRole bool                     `ddl:"static" sql:"DATABASE ROLE"`
+	IfExists     *bool                    `ddl:"keyword" sql:"IF EXISTS"`
+	name         DatabaseObjectIdentifier `ddl:"identifier"`
+}
+
+// showDatabaseRoleOptions is based on https://docs.snowflake.com/en/sql-reference/sql/show-database-roles.
+// At the time of writing LIKE is not visible in the docs, but it works.
+type showDatabaseRoleOptions struct {
+	show          bool                    `ddl:"static" sql:"SHOW"`
+	databaseRoles bool                    `ddl:"static" sql:"DATABASE ROLES"`
+	Like          *Like                   `ddl:"keyword" sql:"LIKE"`
+	in            bool                    `ddl:"static" sql:"IN DATABASE"`
+	Database      AccountObjectIdentifier `ddl:"identifier"`
+}
+
+// databaseRoleDBRow is used to decode the result of a SHOW DATABASE ROLES query.
+type databaseRoleDBRow struct {
+	CreatedOn              string         `db:"created_on"`
+	Name                   string         `db:"name"`
+	IsDefault              sql.NullString `db:"is_default"`
+	IsCurrent              sql.NullString `db:"is_current"`
+	IsInherited            sql.NullString `db:"is_inherited"`
+	GrantedToRoles         sql.NullString `db:"granted_to_roles"`
+	GrantedToDatabaseRoles sql.NullString `db:"granted_to_database_roles"`
+	GrantedDatabaseRoles   sql.NullString `db:"granted_database_roles"`
+	Owner                  string         `db:"owner"`
+	Comment                sql.NullString `db:"comment"`
+	OwnerRoleType          sql.NullString `db:"owner_role_type"`
+}
+
+// DatabaseRole is a user-friendly result for a SHOW DATABASE ROLES query.
+// At the time of writing there is no format specified in the docs.
+type DatabaseRole struct {
+	CreatedOn              string
+	Name                   string
+	IsDefault              bool
+	IsCurrent              bool
+	IsInherited            bool
+	GrantedToRoles         string
```

Comment:
There is no doc for this, but as far as I could see, this will be number of roles, so I will probably change it to integer (but after I confirm it with my following PR with grants)